prettier-maven-plugin/src/main/java/com/hubspot/maven/plugins/prettier/internal/PrettierPatcher.java

```diff
@@ -67,12 +68,12 @@ private void applyPatch(URL patch, Path directory) throws MojoExecutionException
   }
 
   private File copyToFile(URL source) throws IOException {
-    Path tmpFile = Files.createTempFile(originalDirectory.getParent(), "prettier-", ".patch");
-    tmpFile.toFile().deleteOnExit();
+    File tmpFile = Files.createTempFile(originalDirectory.getParent(), "prettier-", ".patch").toFile();
+    tmpFile.deleteOnExit();
 
-    byte[] bytes = Resources.toByteArray(source);
-    Files.write(tmpFile, bytes);
-
-    return tmpFile.toFile();
+    try (FileOutputStream fileOutputStream = new FileOutputStream(tmpFile)) {
+      fileOutputStream.getChannel().transferFrom(Channels.newChannel(source.openStream()), 0, Long.MAX_VALUE);
+    }
```

Comment:
I think this leaks the source `InputStream`. It looks like we can keep everything using `Path` and the newer nio methods with something like:
```java
try (
  ReadableByteChannel inputChannel = Channels.newChannel(source.openStream());
  FileChannel outputChannel = FileChannel.open(tmpFile, StandardOpenOption.WRITE)
) {
  outputChannel.transferFrom(inputChannel, 0, Long.MAX_VALUE);
}
```

Reply:
Nice! This is better in multiple ways.